src/cluster.c

```diff
@@ -2808,11 +2969,23 @@ void clusterSendPing(clusterLink *link, int type) {
         dictReleaseIterator(di);
     }
 
-    /* Ready to send... fix the totlen field and queue the message in the
-     * output buffer. */
-    totlen = sizeof(clusterMsg)-sizeof(union clusterMsgData);
+    
+    int totlen = 0;
+    int extensions = 0;
+    /* Set the initial extension position */
+    clusterMsgPingExt *cursor = getInitialPingExt(hdr, gossipcount);
+    /* Add in the extensions */
+    if (myself->hostname) {
```

Comment:
The max hostname length is not a trivial number, i.e., 256 bytes. The non-stop broadcasting of a node's host name in a large cluster increases the overhead on both network and CPUs when the cluster is in a steady state. This impact can also grow with more extensions added in the future, It will be great if a generic mechanism could be devised such that the broadcasting of these extensions can be optional and only activated when a change is detected in the cluster (for instance, epoch bump or new node arrival).

Reply:
Please open a new issue or PR. Comments on closed PRs are not often read by people.

Reply:
I read them :D